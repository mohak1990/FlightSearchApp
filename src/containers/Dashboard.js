import React, {Fragment} from 'react';
import Dashboard from '../components/dashboard'
import DashboardHeader from '../components/dashboard/header'
import { connect } from 'react-redux'
import { showDetails, selectOneWayFlight, selectReturnFlight } from '../modules/dashboard'


class DashboardContainer extends React.Component {

  render() {
    return (
      <Fragment>
      {this.props.flights.length ? (
        <Fragment>
          <DashboardHeader
            flights = {this.props.flights}
            isReturnFlight = {this.props.isReturnFlight}
            originCity = {this.props.originCity}
            destCity = {this.props.destCity}
            date = {this.props.depDate}
          />
          {this.props.isReturnFlight &&
            <DashboardHeader
              isReturnFlight = {this.props.isReturnFlight}
              flights = {this.props.returnFlights}
              originCity = {this.props.destCity}
              destCity = {this.props.originCity}
              date = {this.props.returnDate}
            />
          }
          <div className = "dashboard--show-flights">
            <Dashboard
             flightDirection = {"One Way"}
             isReturnFlight = {this.props.isReturnFlight}
             selectOneWayFlight = {this.props.selectOneWayFlight}
             selectReturnFlight = {this.props.selectReturnFlight}
             flights = { this.props.flights }
             showDetails = {(i) => this.props.showDetails(i)}
             expandedView = {this.props.expandedView}
             originCity = {this.props.originCity}
             destCity = {this.props.destCity}
             setDepDate = {this.props.depDate}
            />
            {this.props.isReturnFlight &&
              <Dashboard
                flightDirection = {"Return"}
                isReturnFlight = {this.props.isReturnFlight}
                selectReturnFlight = {this.props.selectReturnFlight}
                flights = { this.props.returnFlights }
                showDetails = {(i) => this.props.showDetails(i)}
                expandedView = {this.props.expandedView}
                originCity = {this.props.originCity}
                destCity = {this.props.destCity}
                setReturnDate = {this.props.returnDate}
              />
            }
          </div>
        </Fragment>
      )
      :
      <div className = "dashboard--no-flights">
        Search for Flights
      </div>
      }
      </Fragment>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
    flights: state.dashboard.flights,
    returnFlights: state.dashboard.returnFlights,
    isReturnFlight: state.filterPanel.isReturnFlight,
    originCity: state.filterPanel.originCity,
    destCity: state.filterPanel.destCity,
    depDate: state.filterPanel.depDate,
    returnDate: state.filterPanel.returnDate
})

const mapDispatchToProps = function(dispatch, ownProps){
  return {
   showDetails: (i) => dispatch(showDetails(i)),
   selectOneWayFlight: (i) => dispatch(selectOneWayFlight(i)),
   selectReturnFlight: (i) => dispatch(selectReturnFlight(i))
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DashboardContainer)
