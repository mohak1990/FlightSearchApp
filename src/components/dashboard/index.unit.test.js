import React from 'react';
import Dashboard from './index';
import { mount, shallow  } from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow'

test('Snapshot testing for dashboard component', () => {
  const renderer = new ShallowRenderer()
  const result = renderer.render(<Dashboard flights = {[]} />)
  expect(result).toMatchSnapshot()
});

it('render Dashboard correctly', () => {
    const DashboardComponent = mount(<Dashboard flights = {[]} />).render();
    expect(DashboardComponent.hasClass('dashboard')).toBeTruthy();
});

it('render Dashboard correctly for return flights', () => {
    const DashboardComponent = mount(<Dashboard flights = {[]} isReturnFlight = {true} />).render();
    expect(['dashboard', 'dashboard--returnFlight'].every(c => DashboardComponent.hasClass(c))).toBeTruthy();
});

it('check if FlightDetail component exist', () => {
    const DashboardComponent = shallow(<Dashboard flights = {[{},{}]} />);
    expect(DashboardComponent.find('FlightDetail')).toHaveLength(2);
});
