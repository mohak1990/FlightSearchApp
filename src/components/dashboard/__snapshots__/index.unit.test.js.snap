// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing for dashboard component 1`] = `
<div
  className="dashboard"
>
  <div
    className="dashboard--content"
  />
</div>
`;
